authz/sdk_server_interceptors.go

```diff
@@ -73,3 +82,88 @@ func (i *StaticInterceptor) StreamInterceptor(srv interface{}, ss grpc.ServerStr
 	}
 	return handler(srv, ss)
 }
+
+// FileWatcherInterceptor contains details used to make authorization decisions
+// by watching a file path that contains authorization policy in JSON format.
+type FileWatcherInterceptor struct {
+	internalInterceptor unsafe.Pointer // *StaticInterceptor
+	policyFile          string
+	policyContents      []byte
+	refreshDuration     time.Duration
+	cancel              context.CancelFunc
+}
+
+// NewFileWatcher returns a new FileWatcherInterceptor from a policy file
+// that contains JSON string of authorization policy and a refresh duration to
+// specify the amount of time between policy refreshes.
+func NewFileWatcher(file string, duration time.Duration) (*FileWatcherInterceptor, error) {
+	if file == "" {
+		return nil, fmt.Errorf("authorization policy file path is empty")
+	}
+	if duration <= time.Duration(0) {
+		return nil, fmt.Errorf("requires refresh interval(%v) greater than 0s", duration)
+	}
+	i := &FileWatcherInterceptor{policyFile: file, refreshDuration: duration}
+	if err := i.updateInternalInterceptor(); err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	i.cancel = cancel
+	// Create a background go routine for policy refresh.
+	go i.run(ctx)
+	return i, nil
+}
+
+func (i *FileWatcherInterceptor) run(ctx context.Context) {
+	ticker := time.NewTicker(i.refreshDuration)
+	for {
+		err := i.updateInternalInterceptor()
+		logger.Infof("authorization policy reload status err: %v", err)
```

Comment:
Should this be a `Warning`?  Is it `INFO` in other languages?

Reply:
By logging on every reload, we can inform the user when the file watcher interceptor recovers from error, like on replacing invalid policy with a valid policy.

In C++, if sdk tracer logs are enabled, we will see INFO logs on every reload.
https://github.com/grpc/grpc/pull/26779
Since the PR is under review, I added a comment at the log statement for your reference.

Reply:
It's common to disable INFO logs.

My preference would be:
- On reload error, log at warning level
- On reload success, don't log anything

It looks like the reload will currently be logged even if there was no change in value - at least in the Go PR - meaning this will just be continuous log spam.  If there are changes upon reloading, it could possibly be useful.

Reply:
Thank you @dfawley for the suggestion. Updated the logging to log at warning level for errors. Additionally included logs at Info level, when loading new policy. PTAL.